ddtrace/mocktracer/mockspan.go

```diff
@@ -7,282 +7,122 @@ package mocktracer // import "github.com/DataDog/dd-trace-go/v2/ddtrace/mocktrac
-	finishTime   time.Time
-	finished     bool
+func MockSpan(s *tracer.Span) *Span {
+	return &Span{Span: s}
+}
 
-	startTime time.Time
-	parentID  uint64
-	context   *spanContext
-	tracer    *mocktracer
+func (s *Span) OperationName() string {
+	return s.Name
 }
 
-// SetTag sets a given tag on the span.
-func (s *mockspan) SetTag(key string, value interface{}) {
-	s.Lock()
-	defer s.Unlock()
-	if s.finished {
-		return
+func (s *Span) Tag(k string) interface{} {
+	if s == nil {
+		return nil
 	}
-	if s.tags == nil {
-		s.tags = make(map[string]interface{}, 1)
+	switch k {
+	case ext.SpanName:
+		return s.Name
+	case ext.ServiceName:
+		return s.Service
+	case ext.ResourceName:
+		return s.Resource
+	case ext.SpanType:
+		return s.Type
 	}
-	if key == ext.SamplingPriority {
-		switch p := value.(type) {
-		case int:
-			s.context.setSamplingPriority(p)
-		case float64:
-			s.context.setSamplingPriority(int(p))
+	if s.Meta != nil {
+		if r, ok := s.Meta[k]; ok {
+			return r
 		}
 	}
-	s.tags[key] = value
-}
-
-func (s *mockspan) FinishTime() time.Time {
-	s.RLock()
-	defer s.RUnlock()
-	return s.finishTime
-}
-
-func (s *mockspan) StartTime() time.Time { return s.startTime }
-
-func (s *mockspan) Tag(k string) interface{} {
-	s.RLock()
-	defer s.RUnlock()
-	return s.tags[k]
-}
-
-func (s *mockspan) Tags() map[string]interface{} {
-	s.RLock()
-	defer s.RUnlock()
-	// copy
-	cp := make(map[string]interface{}, len(s.tags))
-	for k, v := range s.tags {
-		cp[k] = v
+	if s.Metrics != nil {
+		if r, ok := s.Metrics[k]; ok {
+			return r
+		}
 	}
-	return cp
-}
-
-func (s *mockspan) TraceID() uint64 { return s.context.traceID }
-
-func (s *mockspan) SpanID() uint64 { return s.context.spanID }
-
-func (s *mockspan) ParentID() uint64 { return s.parentID }
-
-func (s *mockspan) OperationName() string {
-	s.RLock()
-	defer s.RUnlock()
-	return s.name
-}
-
-// SetOperationName resets the original operation name to the given one.
-func (s *mockspan) SetOperationName(operationName string) {
-	s.Lock()
-	defer s.Unlock()
-	s.name = operationName
-	return
-}
-
-// BaggageItem returns the baggage item with the given key.
-func (s *mockspan) BaggageItem(key string) string {
-	return s.context.baggageItem(key)
+	return nil
 }
 
-// SetBaggageItem sets a new baggage item at the given key. The baggage
-// item should propagate to all descendant spans, both in- and cross-process.
-func (s *mockspan) SetBaggageItem(key, val string) {
-	s.context.setBaggageItem(key, val)
-	return
-}
-
-// Finish finishes the current span with the given options.
-func (s *mockspan) Finish(opts ...ddtrace.FinishOption) {
-	var cfg ddtrace.FinishConfig
-	for _, fn := range opts {
-		fn(&cfg)
-	}
-	var t time.Time
-	if cfg.FinishTime.IsZero() {
-		t = time.Now()
-	} else {
-		t = cfg.FinishTime
-	}
-	if cfg.Error != nil {
-		s.SetTag(ext.Error, cfg.Error)
+func (s *Span) Tags() map[string]interface{} {
+	r := make(map[string]interface{})
+	if s == nil {
+		return r
 	}
```

Comment:
I'd have allocated the map _after_ the nil-check, so that the map can be pre-allocated to the correct size.